src/polycap.c

```diff
@@ -780,42 +773,97 @@ void count(struct mumc *absmu, struct inp_file *cap, int *icount, struct cap_pro
+		c = (cap->d_screen - calc->rh[2]) / calc->v[2];
+		xp = calc->rh[0] + c*calc->v[0];
+		yp = calc->rh[1] + c*calc->v[1];
 
-		delta_traj[0] = c*calc[thread_id].v[0];
-		delta_traj[1] = c*calc[thread_id].v[1];
-		delta_traj[2] = c*calc[thread_id].v[2];
+		delta_traj[0] = c*calc->v[0];
+		delta_traj[1] = c*calc->v[1];
+		delta_traj[2] = c*calc->v[2];
 		ds = sqrt(scalar(delta_traj,delta_traj));
-		calc[thread_id].traj_length = calc[thread_id].traj_length + ds;
+		calc->traj_length = calc->traj_length + ds;
 
 		ind_x = (int)floor(xp/profile->binsize)+NSPOT/2;
 		ind_y = (int)floor(yp/profile->binsize)+NSPOT/2;
 		if(ind_x < NSPOT && ind_x >= 0){
 			if(ind_y < NSPOT && ind_y >= 0){
-				leaks->spot[ind_x][ind_y] = leaks->spot[ind_x][ind_y] + calc[thread_id].w[0];
+				leaks->spot[ind_x][ind_y] = leaks->spot[ind_x][ind_y] + calc->w[0];
 				}
 			}
 
-		if(*icount <= IMSIZE-1){
-			imstr[*icount].xm = yp;
-			imstr[*icount].ym = xp;
-			imstr[*icount].xm1 = calc[thread_id].v[1];
-			imstr[*icount].ym1 = calc[thread_id].v[0];
-			imstr[*icount].warr = calc[thread_id].w[0];
+		if(icount <= IMSIZE-1){
+			imstr[icount].xm = yp;
+			imstr[icount].ym = xp;
+			imstr[icount].xm1 = calc->v[1];
+			imstr[icount].ym1 = calc->v[0];
+			imstr[icount].warr = calc->w[0];
 			}
 		} //if(dp1 > hex_edge_dist || dp2 > hex_edge_dist || dp3 > hex_edge_dist) ... else ...
 	return;
 	}
+
+// ---------------------------------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------------------------
+
+struct calcstruct *init_calcstruct(unsigned long int seed, struct cap_profile *profile, struct mumc *absmu) {
+	const polycap_rng_type *T = polycap_rng_mt19937; //Mersenne twister rng
+	struct calcstruct *calc = malloc(sizeof(struct calcstruct));
+	int j;
+
+	/*give arrays inside calc struct appropriate dimensions*/
+	calc->sx = malloc(sizeof(double)*(profile->nmax+1));
+	if(calc->sx == NULL){
+		printf("Could not allocate calc->sx memory.\n");
+		exit(0);
+	}
+	calc->sy = malloc(sizeof(double)*(profile->nmax+1));
+	if(calc->sy == NULL){
+		printf("Could not allocate calc->sy memory.\n");
+		exit(0);
+	}
+	calc->absorb = malloc(sizeof(double)*(profile->nmax+1));
+	if(calc->absorb == NULL){
+		printf("Could not allocate calc->absorb memory.\n");
+		exit(0);
+	}
+	calc->w = malloc(sizeof(double)*(absmu->n_energy+1));
+	if(calc->w == NULL){
+		printf("Could not allocate calc->w memory.\n");
+		exit(0);
+	}
+	calc->cnt = malloc(sizeof(double)*(absmu->n_energy+1));
+	if(calc->cnt == NULL){
+		printf("Could not allocate calc->cnt memory.\n");
+		exit(0);
+	}
+	/*copy correct values into corresponding calc struct variable*/
+	calc->istart = 0;
+	calc->ienter = 0;
+	calc->iesc = 0;
+	calc->ix = 0;
+	for(j=0; j<=profile->nmax; j++){
+		calc->sx[j] = profile->arr[j].sx;
+		calc->sx[j] = profile->arr[j].sy; // -> very fishy!!! assign to calc->sy[j]???
```

Comment:
@PieterTack Kan je eens controleren wat hier aan de hand is? Lijkt me verdacht...
De enige reden dat dit toch werkt is dat `calc->sx` en `calc->sy` overschreven worden in `start()`. Eigenlijk worden `profile->arr[j].sx` en `profile->arr[j].sy` nooit gebruikt volgens mij...

Reply:
@tschoonj Je hebt gelijk dat lijn 854 moet assigned worden aan calc->sy[j]
Verder heb je ook gelijk dat die profile->arr[j].sx en.sy blijkbaar nooit gebruikt worden. Was me ontgaan toen ik Laszlo's code herschreef. Technisch gezien zouden ze de PC central axis coordinaten voorstellen (momenteel gegeven door de .axs file en voorlopig altijd gelijk aan 0). 
Wanneer ik dan de aanpassingen heb gemaakt waarbij de .prf, .ext en .axs files niet meer moeten meegegeven worden zal ik in start() eens moeten terugkijken om te zorgen dat de orientatie van het PC juist wordt meegegeven (in het geval het PC niet mooi on-axis ligt)

Reply:
Ok bedankt voor er eens snel naar te kijken. Zal het voorlopig zo laten dan.

Het volgende waar ik eens ga kijken zijn die do-while loops met hun `calc->iesc`condities. Eens zien of dat ook kan met gewone for loops.

Ook calcstruct ga ik nog eens herbekijken... ik zou dat graag hervormen naar een struct die alle properties van een enkel foton bevat.

Gelukkig nieuwjaar trouwens!

Reply:
Uiteraard ook voor jou een schitterend nieuwjaar!
Verander je wel nog lijn 854 naar ...->sy[j]? Da's wel een belangrijke eigenlijk. (beetje vreemd dat mijn simulaties toch dezelfde resultaten gaven als Laszlo's oude code :P)

Voor je die andere dingen bekijkt, wil je dan eerst eens melden wanneer ik al kan mergen met de huidige veranderingen van hoe calc gedeeld wordt over de threads enzo? Ben bang dat als ik anders al aan mijn aanpassingen begin dat we met merge problemen gaan komen te zitten (daar ik nog met de oude manier van calc zou zitten, en jij al met de nieuwe)

Die do naar for loops wil ik ook gerust bekijken hoor. Hoe bespreken we overigens wie wat aanpakt? Kwestie dat ik niet plots het gevoel heb dat ik niks meer deftig kan bijdragen omdat jij veel sneller en efficienter programmeert :P

Reply:
Ok, zonet de commit aangepast. Van mij mag je mergen zodra de CI builds groen zijn.

Belangrijk was het niet echt, want zoals eerder gezegd werden die waarden toch overschreven in `start()` alvorens ze echt gebruikt werden.

Merge conflicten zijn normaal geen probleem: als jouw pull request niet kan gemerged worden nadat een andere pull request net gemerged is, moet je lokaal het volgende doen:

````bash
git fetch
git rebase origin/master
# indien merge problemen -> fix ze, gevolgd door
git rebase --continue
````
Als er dan nog merge conflicten zijn, fix ze, altijd gevolgd door `git rebase --continue`, tot alles in orde is.

Daarna moet je je branch weer naar github pushen, dit keer wel met -f omdat de geschiedenis herschreven is...

````bash
git push -f
````

Laat maar weten als je nog vragen hebt over git, of vraag uw collega Stijn eventueel. Sorry, had u beter de laatste keer wat verteld over `git rebase`, aangezien dat toch iets is dat je vaak moet gebruiken, zeker met onze workflow.

Ik zal nu even stoppen met code te schrijven en eerst werken aan een voorstel tot API die in de header kan geplaatst worden. Dat zal een goed idee geven hoe de code moet geherstructureerd worden.

Zal daarvoor een aparte pull request maken.

Werk intussen maar aan wat je wil, maar open een pull-request voor je branch van zodra je een commit hebt, dan kan de CI werken en kan ik kijken wat ge aan het doen zijt 😄